synapse/replication/tcp/streams/_base.py

```diff
@@ -153,61 +162,62 @@ def discard_updates_and_advance(self):
         """
         self.last_token = self.current_token()
 
-    async def get_updates(self):
+    async def get_updates(self) -> Tuple[List[Tuple[int, JsonDict]], int, bool]:
         """Gets all updates since the last time this function was called (or
         since the stream was constructed if it hadn't been called before).
 
         Returns:
-            Deferred[Tuple[List[Tuple[int, Any]], int]:
-                Resolves to a pair ``(updates, current_token)``, where ``updates`` is a
-                list of ``(token, row)`` entries. ``row`` will be json-serialised and
-                sent over the replication steam.
+            Resolves to a pair `(updates, new_last_token, limited)`, where
```

Comment:
"Resolves" is more appropriate to deferreds:

```suggestion
            A triplet `(updates, new_last_token, limited)`, where
```